user/super/com/google/gwt/emul/java/util/Objects.java

```diff
@@ -129,6 +129,37 @@ public static <T> T requireNonNull(T obj, Supplier<String> messageSupplier) {
     return obj;
   }
 
+  public static <T> T requireNonNullElse(T obj, T defaultObj) {
+    return obj != null ? obj : requireNonNull(defaultObj);
+  }
+
+  public static <T> T requireNonNullElseGet(T obj, Supplier<? extends T> supplier) {
+    return obj != null ? obj : requireNonNull(supplier.get());
+  }
+
+  public static int checkIndex(int index, int length) {
+    if (index < 0 || index >= length) {
+      throw new IndexOutOfBoundsException("Index " + index + " out of bounds for length " + length);
+    }
+    return index;
+  }
+
+  public static int checkFromToIndex(int fromIndex, int toIndex, int length) {
+    if (fromIndex < 0 || fromIndex > toIndex || toIndex > length) {
+      throw new IndexOutOfBoundsException("Range [" + fromIndex + ", " + toIndex
+          + ") out of bounds for length " + length);
+    }
+    return fromIndex;
+  }
+
+  public static int checkFromIndexSize(int fromIndex, int size, int length) {
+    if (fromIndex < 0 || size < 0 || fromIndex + size > length) {
```

Comment:
The JavaDoc indicates that `fromIndex + size > length` should take integer overflow into account. Maybe add a comment that JS max integer is (2^53 – 1) and thus adding any two Java integers will not overflow.